docs/build.md

```diff
@@ -0,0 +1,8 @@
+* Build custom software
+
+The build code keeps a cache of all packages source code (Git/Mercurial/SVN)
+repositories so that they do not need to be downloaded on every build. If it is
+desired to change the source code of a package, you may edit the package local
+repository content. To build the package from this modified local repository,
+either use a file:// URL pointing to it in the package metadata (YAML)
+file or disable package repositories update in config/host_os.yaml.
```

Comment:
Yes, I believe the recommended way would be to have the package source code cloned. Not necessarily again, because the user might have a custom package repository even before executing their first Host OS build. The new [naive] user should not need to know where we keep the cached repositories and how we manage them.

The only usefulness of this is being able to make quick changes to sources to test something, without having to commit or clone to another location. I've never used it myself, but I see it as a nice feature for some use cases.